src/control/server/drpc.go

```diff
@@ -44,21 +47,32 @@ func getDrpcClientConnection(sockDir string) *drpc.ClientConnection {
 	return drpc.NewClientConnection(clientSock)
 }
 
+func createSocketDir(sockDir string) error {
+	// TODO: decide whether we want to bail if it doesn't exist, might fall under
+	// category of items that should be created during configurations management
+	// as locations may not be user creatable. If we do detect, verify permissions
+	// and that it is indeed a directory and not a file.
```

Comment:
I had forgotten this logic was here. In the long term, I would lean toward your suggestion that these socket directories should be created during configuration management.

Reply:
I agree as well. Lets remove the socket directory creation code and rely on systemd unit files to do it for us.